src/turbopuffer.ts

```diff
@@ -131,25 +145,41 @@ export class Turbopuffer {
       requestBody = JSON.stringify(body);
     }
 
-    const response = await fetch(url.toString(), {
-      method,
-      headers,
-      body: requestBody,
-    });
-    if (response.status >= 400) {
-      let message = response.statusText;
-      try {
-        let body = await response.text();
-        if (body) {
-          message = body;
-        }
-      } catch (_: any) {}
-      throw new TurbopufferError(message, { status: response.status });
+    const maxAttempts = retryable ? 3 : 1;
+    var response!: Response;
+    var error: TurbopufferError | null = null;
+    for (let attempt = 0; attempt < maxAttempts; attempt++) {
+      response = await fetch(url.toString(), {
+        method,
+        headers,
+        body: requestBody,
+      });
+      if (response.status >= 400) {
+        let message = response.statusText;
+        try {
+          let body = await response.text();
```

Comment:
we often return 400 with a json response I think, maybe we can look at the content-type response header to decide whether to parse json or not

Reply:
That does work, but the error will end up being a json string, less readable than displaying the actual error message in the "error" field of the parsed json.

Reply:
Ahh, fair — will make an attempt to get the error here